File: leetcode/tree/230.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int result;
    int count = 0;
    int kthSmallest(TreeNode* root, int k) {
        count = k;
        traverse(root);
        return result;
    }
    void traverse(TreeNode* root) {
        if ( root == nullptr ) {
            return;
        }
        traverse(root->left);
        --count;
        if ( count == 0 ) {
            result = root->val;
            return;
        }
        traverse(root->right);
    }
};

File: leetcode/tree/230.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @param {number} k
 * @return {number}
 */
var kthSmallest = function(root, k) {
  var result;
  var count = 0;
  var bruh = (root) => {
      if ( root === null ) {
          return;
      }
      bruh(root.left);
      
      if ( count <= k ) {
          count++;
      }
      if ( count === k ) {
          result = root.val;
      } else if ( count > k ) {
          return;
      }
      bruh(root.right);
  };
  bruh(root);
  return result;
};